packages/rum-core/src/domain/foregroundContexts.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Configuration,
+  noop,
+  addEventListener,
+  DOM_EVENT,
+  RelativeTime,
+  elapsed,
+  relativeNow,
+  Duration,
+  addMonitoringMessage,
+  toServerDuration,
+} from '@datadog/browser-core'
+import { InForegroundPeriod } from '../rawRumEvent.types'
+
+// Arbitrary value to cap number of element (mostly for backend)
+const MAX_NUMBER_OF_FOCUSED_TIME = 500
+
+export interface ForegroundContexts {
+  getInForeground: (startTime: RelativeTime) => boolean | undefined
+  getInForegroundPeriods: (startTime: RelativeTime, duration: Duration) => InForegroundPeriod[] | undefined
+  stop: () => void
+}
+
+export interface ForegroundPeriod {
+  start: RelativeTime
+  end?: RelativeTime
+}
+
+let foregroundPeriods: ForegroundPeriod[] = []
+
+export function startForegroundContexts(configuration: Configuration): ForegroundContexts {
+  if (!configuration.isEnabled('track-foreground')) {
+    return {
+      getInForeground: () => undefined,
+      getInForegroundPeriods: () => undefined,
+      stop: noop,
+    }
+  }
+  if (document.hasFocus()) {
+    addNewForegroundPeriod()
+  }
+
+  const { stop: stopForegroundTracking } = trackFocus(addNewForegroundPeriod)
+  const { stop: stopBlurTracking } = trackBlur(closeForegroundPeriod)
+  return {
+    getInForeground,
+    getInForegroundPeriods,
+    stop: () => {
+      foregroundPeriods = []
+      stopForegroundTracking()
+      stopBlurTracking()
+    },
+  }
+}
+
+function addNewForegroundPeriod() {
+  if (foregroundPeriods.length > MAX_NUMBER_OF_FOCUSED_TIME) {
+    addMonitoringMessage('Reached maximum of foreground time')
+    return
+  }
+  const currentForegroundPeriod = foregroundPeriods[foregroundPeriods.length - 1]
+  if (currentForegroundPeriod != null && currentForegroundPeriod.end === undefined) {
```

Comment:
IMO comparing `currentForegroundPeriod` to `null` is counter-intuitive, since `currentForegroundPeriod` is never `null`. So either using its falsysness or comparing it to `undefined` seems find to me.

I don't think there is a strict convention on this, but I've never used `==` in my career, this operator scares me :D

Reply:
> I don't think there is a strict convention on this, but I've never used == in my career, this operator scares me :D

Using it with `null` is fine. I always do that.
Since eslint is okay with that we should add a eslint rule to remove it so we don't have to check for that in review.

Falsy check scares me a lot more than `== null`

Reply:
agree on adding an eslint rule to avoid those discussions.

the "convention" here is that we don't use `==` and `null`.

If you are not fond of falsy checks, that should work:
```suggestion
  if (currentForegroundPeriod !== undefined && currentForegroundPeriod.end === undefined) {
```